template/.github/workflows/ci.yml

```diff
@@ -170,3 +170,60 @@ jobs:
 
     - name: Run the functional test suite
       run: poetry run invoke test --suite=functional
+
+build-and-test-docker-image:
+    name: Build and Test Docker Image
+    needs: [static-analysis-checks, type-checking, run-system-tests, run-functional-tests]
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:14
+        env:
+          POSTGRES_USER: dev
+          POSTGRES_PASSWORD: dev_password
+          POSTGRES_DB: dev
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+            # Allows us to use the same env config used locally by mapping to port to
+            # the test runner container.
+            - 5432:5432
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+
+    - name: Install dependencies
+      run: |
+        pip install invoke
+        cp example.env .env
+
+    - name: Modify DATABASE_URL for GitHub Actions
+      run: |
+        echo 'DATABASE_URL=postgres://dev:dev_password@postgres:5432/dev'  >> .env
+        echo 'DJANGO_SETTINGS_MODULE=avocado.main.settings' >> .env
+        cat .env
+
+    - name: Build Docker Image
+      run: invoke build-image --tag abc
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** ‏Do not set the `tag`. 

The `build-image` task performs a series of operations to determine the correct tag to apply for this CI run based on the branch and commit. This ensure that each build has a unique, and re-usage tag we can reference back to the git commit should we have problems.

The general build and deployment process should be:

- build image with unique, commit based tag
- publish image to a docker registry

(if working with a test env)
- deploy that tag to test environment
- ensure test environment is working
- release the _same_ tag to production.

(if working with continuous delivery)
- deploy same tagged image to all environments

Reply:
I passed `tag` here so we can run the created image in the next ci step (As we might not know the related tag to run the image). So, do we need an invoke task to run the image?

Reply:
Ultimately the build step should be a precursor to publishing the image to a registry. So it needs to use the calculated tag.

You could either:

- do a task that runs the container, so it can use the same function to generate the tag
- write a task that just prints the calculated tag. Assign the output from that to an envar and then use it in your docker run

Reply:
I added a task to run the docker image.